### PR TITLE
fix: histogram improvement

### PR DESCRIPTION
**Pull request by emori-ctc**
## Pre-review checklist for the PR author

- [ ] I've confirmed the [contribution guidelines](https://github.com/tier4/caret/blob/main/.github/CONTRIBUTING.md).
- [ ] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR has been properly tested.
- [x] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] (Optional) The PR has been properly tested with CARET_report verification.
- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

File: src/caret_analyze/plot/visualize_lib/bokeh/bokeh.py [hunk 1]
```diff
@@ -269,23 +272,44 @@ def histogram(
...
-            plot.add_tools(hover)
+        hists = []
+
+        for hist_type in data_list:
+            hist, bins = histogram(hist_type, 20, data_range, density=False)
+            hists.append(hist)
+
+        hists_t = np.array(hists).T
+
+        colors = [
```
Comment (rokamu623) [inline src/caret_analyze/plot/visualize_lib/bokeh/bokeh.py:283]: Please use `ColorSelector` (`ColorSelectorFactory`) to determine the color of the graph.

File: src/caret_analyze/plot/visualize_lib/bokeh/bokeh.py [hunk 2]
```diff
@@ -269,23 +272,44 @@ def histogram(
...
-                             color=color_selector.get_color())
-            legend_manager.add_legend(target_object, quad)
-            hover = HoverTool(
-                tooltips=[(x_label, '@left'), ('The number of samples', '@top')], renderers=[quad]
-                )
-            plot.add_tools(hover)
+        hists = []
+
+        for hist_type in data_list:
+            hist, bins = histogram(hist_type, 20, data_range, density=False)
```
Comment (rokamu623) [inline src/caret_analyze/plot/visualize_lib/bokeh/bokeh.py:278]: Please make the variable names easy to understand.

e.g. `data`, `data_by_one_object`, `data_row`

File: src/caret_analyze/plot/visualize_lib/bokeh/bokeh.py [hunk 3]
```diff
@@ -269,23 +272,44 @@ def histogram(
...
+            for _ in range(len(hists_t))
+            ]
+
+        quad_dicts = {}
+
+        for i, h in enumerate(hists_t):
+            data = list(zip(h, colors, target_objects))
+            data.sort(key=lambda x: x[0], reverse=True)
+            for top, col, target_object in data:
+                quad = plot.quad(
```
Comment (rokamu623) [inline src/caret_analyze/plot/visualize_lib/bokeh/bokeh.py:294]: Is it possible to make the `legend` of a histogram with no data (all elements are 0) transparent?

If you can't come up with an idea, I can support you.

**Review by rokamu623 — CHANGES_REQUESTED** [review]
In particular, please be sure to address any changes regarding color.

File: src/caret_analyze/plot/visualize_lib/bokeh/bokeh.py [hunk 4]
```diff
@@ -269,23 +268,45 @@ def histogram(
...
+            hist, bins = histogram(hist_type, 20, data_range, density=False)
+            hists.append(hist)
+
+        hists_t = np.array(hists).T
+
+        colors = []
+
+        color_selector = ColorSelectorFactory.create_instance('unique')
+        for _ in hists_t:
+            colors.append(color_selector.get_color())
```
Comment (rokamu623) [inline src/caret_analyze/plot/visualize_lib/bokeh/bokeh.py:283]: How about this implementation?
```suggestion
        color_selector = ColorSelectorFactory.create_instance('unique')
        colors = [color_selector.get_color() for _ in hists_t]
```

**Review by rokamu623 — CHANGES_REQUESTED** [review]

File: src/caret_analyze/plot/visualize_lib/bokeh/bokeh.py [hunk 5]
```diff
@@ -269,23 +268,41 @@ def histogram(
...
+        for hist_type in data_list:
+            hist, bins = histogram(hist_type, 20, data_range, density=False)
+            hists.append(hist)
+
+        hists_t = np.array(hists).T
+
+        color_selector = ColorSelectorFactory.create_instance('unique')
+        colors = [color_selector.get_color() for _ in hists_t]
+
+        quad_dicts = {t: [] for t in target_objects}
```
Comment (rokamu623) [inline src/caret_analyze/plot/visualize_lib/bokeh/bokeh.py:282]: There was an annotation omission in my suggestion.
```suggestion
        quad_dicts: dict[HistTypes, list[GlyphRenderer]] = {t: [] for t in target_objects}
```

**Review by rokamu623 — CHANGES_REQUESTED** [review]

**Review by bopeng-saitama — APPROVED** [review]
LGTM

**Review by xygyo77 — DISMISSED** [review]
image

LGTM

**Review by rokamu623 — APPROVED** [review]
LGTM!
and it is confirmed to works properly.
image

**Pull request merged**